lib/statsd/instrument.rb

```diff
@@ -130,6 +131,12 @@ def statsd_count_success(method, name, sample_rate: nil, tags: nil, no_prefix: f
           client ||= StatsD.singleton_client
           suffix = truthiness == false ? "failure" : "success"
           key = StatsD::Instrument.generate_metric_name(name, self, *args)
+
+          if tag_error_class && error
+            tags = tags.dup || {}
+            tags[:error_class] = error.class.name
```

Comment:
I don't know how big a deal it is, but this assumes `tags` is a `Hash`. Most of the time it's true, but it's (badly) documented as `@param [Hash<Symbol, String>, Array<String>] tags (default: nil)`

https://github.com/Shopify/statsd-instrument/blob/a13bc0499bdf9fb4be8fb7894125e8ab8aba489f/lib/statsd/instrument/client.rb#L198

But I guess it would be a pain to support all types.